open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/common/idempotency/IdempotencyValidator.java

```diff
@@ -18,60 +18,59 @@
+     * the consent will be retrieved. Finally following conditions will be validated.
+     *  - Whether the client id sent in the request and client id retrieved from the database are equal
+     *  - Whether the difference between two dates is less than the configured time
+     *  - Whether payloads are equal
      *
-     * @param idempotencyKeyName    Idempotency Key Name
-     * @param idempotencyKeyValue    Idempotency Key Value
-     * @param request                Request Payload
+     * @param consentManageData            Consent Manage Data
      * @return  IdempotencyValidationResult
      */
-    public static IdempotencyValidationResult validateIdempotency(String idempotencyKeyName, String idempotencyKeyValue,
-                                                                  String request, String clientId)
-            throws ConsentManagementException {
-
-        if (Boolean.parseBoolean((configs.get(IdempotencyConstants.IDEMPOTENCY_IS_ENABLED)).toString())) {
-            // If idempotency key name, value or request is empty then cannot proceed with idempotency validation
-            if (idempotencyKeyName == null || idempotencyKeyValue == null || request.isEmpty() || clientId == null) {
-                log.debug("Idempotency Key , NameValue, Client ID or Request is empty. Hence cannot proceed with " +
+    public IdempotencyValidationResult validateIdempotency(ConsentManageData consentManageData)
+            throws IdempotencyValidationException {
+
+        if (IdempotencyValidationUtils.isIdempotencyEnabledFromConfig()) {
+            // If idempotency key value, client id or request is empty then cannot proceed with idempotency validation
+            if (!IdempotencyValidationUtils.isMandatoryParamsPresent(consentManageData,
+                    getIdempotencyHeaderName())) {
+                log.error("Idempotency Key Value, Client ID or Request is empty. Hence cannot proceed with " +
                         "idempotency validation");
                 return new IdempotencyValidationResult(false, false, null, null);
             }
             try {
+                String idempotencyKeyName = getIdempotencyAttributeName(consentManageData.getRequestPath());
+                String idempotencyKeyValue = consentManageData.getHeaders().get(getIdempotencyHeaderName());
                 // Retrieve consent ids that have the idempotency key name and value as attribute
-                ArrayList<String> consentIds = getConsentIdsFromIdempotencyKey(idempotencyKeyName,
-                        idempotencyKeyValue);
+                ArrayList<String> consentIds = IdempotencyValidationUtils
+                        .getConsentIdsFromIdempotencyKey(idempotencyKeyName, idempotencyKeyValue);
                 // Check whether the consent id list is not empty. If idempotency key exists in the database then
                 // the consent Id list will be not empty.
                 if (!consentIds.isEmpty()) {
```

Comment:
Let's do a debug enables check for L77

Reply:
Fixed in https://github.com/wso2/financial-open-banking/pull/51/commits/e1205691228fc4b76853a0eab7dd99dac2516661